### PR TITLE
Add descriptions to all projectiles and warheads

### PR DESCRIPTION
**Pull request by PunkPun**
I add definitions to all projectiles and warheads, even if some of them don't need it.

This PR has 2 goals: 
- Improve documentation
- Improve the formatting of our docks page by making sure all elements have descriptions

### CODE REVIEW

File: OpenRA.Mods.Common/Projectiles/AreaBeam.cs [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 namespace OpenRA.Mods.Common.Projectiles
 {
+	[Desc("Beam projectile that travels straight line.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Projectiles/AreaBeam.cs]:
```suggestion
	[Desc("Beam projectile that travels in a straight line.")]
```

File: OpenRA.Mods.Common/Projectiles/Bullet.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@
 
 namespace OpenRA.Mods.Common.Projectiles
 {
+	[Desc("Projectile that travels straight line.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Projectiles/Bullet.cs]:
```suggestion
	[Desc("Projectile that travels in a straight line.")]
```

File: OpenRA.Mods.Common/Warheads/SpreadDamageWarhead.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@ namespace OpenRA.Mods.Common.Warheads
 {
 	public enum DamageCalculationType { HitShape, ClosestTargetablePosition, CenterPosition }
 
+	[Desc("Apply damage.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Warheads/SpreadDamageWarhead.cs]:
```suggestion
	[Desc("Apply damage in a specified range.")]
```
?

File: OpenRA.Mods.Common/Warheads/LeaveSmudgeWarhead.cs [hunk 4]
```diff
@@ -18,6 +18,7 @@
 
 namespace OpenRA.Mods.Common.Warheads
 {
+	[Desc("Spawn a permanent image.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Warheads/LeaveSmudgeWarhead.cs]: It's not really spawning an image but interacting with the `SmudgeLayer` by adding a smudge to it.

File: OpenRA.Mods.Common/Warheads/HealthPercentageDamageWarhead.cs [hunk 5]
```diff
@@ -15,6 +15,7 @@
 
 namespace OpenRA.Mods.Common.Warheads
 {
+	[Desc("Apply damage based on targets health.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Warheads/HealthPercentageDamageWarhead.cs]:
```suggestion
	[Desc("Apply damage based on the target's health.")]
```

File: OpenRA.Mods.Common/Warheads/FireClusterWarhead.cs [hunk 6]
```diff
@@ -17,6 +17,7 @@
 
 namespace OpenRA.Mods.Common.Warheads
 {
+	[Desc("Fires weapons from the point of contact.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Warheads/FireClusterWarhead.cs]:
```suggestion
	[Desc("Fires weapons from the point of impact.")]
```

File: OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs [hunk 7]
```diff
@@ -19,6 +19,7 @@
 
 namespace OpenRA.Mods.Common.Warheads
 {
+	[Desc("Spawn a temporary image.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs]: The description feels a bit weird. Also, fwiw, it allows to play sounds as well.

File: OpenRA.Mods.Cnc/Projectiles/TeslaZap.cs [hunk 8]
```diff
@@ -17,6 +17,7 @@
 
 namespace OpenRA.Mods.Cnc.Projectiles
 {
+	[Desc("Instant projectile used to create electricity-like effects.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Cnc/Projectiles/TeslaZap.cs]:
```suggestion
	[Desc("Instant-hit projectile used to create electricity-like effects.")]
```

File: OpenRA.Mods.Common/Projectiles/GravityBomb.cs [hunk 9]
```diff
@@ -17,6 +17,7 @@
 
 namespace OpenRA.Mods.Common.Projectiles
 {
+	[Desc("Projectile with flexible physics customisation.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Projectiles/GravityBomb.cs]: "flexible physics customisation" could be a bit more descriptive imo. Something along the lines of "Projectiles with customizable movement and acceleration vector"?

File: OpenRA.Mods.Common/Warheads/LeaveSmudgeWarhead.cs (hunk 4 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Warheads/LeaveSmudgeWarhead.cs]: I'm not really familiar with SmudgeLayer. Should the description be "Interacts with `SmudgeLayer`."

Comment (abcdefg30) [reply OpenRA.Mods.Common/Warheads/LeaveSmudgeWarhead.cs]: Maybe just something like "Creates terrain smudges on impact."

Comment (PunkPun) [reply OpenRA.Mods.Common/Warheads/LeaveSmudgeWarhead.cs]: I'm going with "Creates a smudge in `SmudgeLayer`."

File: OpenRA.Mods.Common/Projectiles/GravityBomb.cs (hunk 9 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Projectiles/GravityBomb.cs]: I'm going with "Projectile with customisable acceleration vector"

File: OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs (hunk 7 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs]: I'm going with "Spawn a sprite with sound."

**Review by abcdefg30 — APPROVED** [review]
Ok

**Review by dragunoff — APPROVED** [review]

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/965e373b57ad93f45748159c497df56f6159851f...9e52ef303ff31e7b5097676430061856f58ac010)

**Pull request merged**